### ISSUE TITLE
[BUG] Failed To Unpersist erros

### ISSUE THREAD

**Issue by pkosyakov** [acoppes/unity-history-window#42]
When in Hierarchy second scene is added and object from this scene was selected while PlayMode, on exit from play mode unity throws Failed To Unpersist errors on object and components which were selected and destroy them. Scene becomes corrupted. Without Gemserk history window active all works Ok

**Comment by pkosyakov** [comment]
As a workaround I added                
```
        private static void SelectionRecorder()
        {
            if (Selection.activeObject != null)
            {
                if (debugEnabled)
                {
                    Debug.Log("Recording new selection: " + Selection.activeObject.name);
                }
                if (EditorApplication.isPlaying)
                {
                    return;
                }
```
Another useful thing to show Hierarchy Objects option in preferences

**Comment by acoppes** [comment]
Will check it, thanks for the report and sorry for having a bug like that.

**Comment by acoppes** [comment]
Ok, I uploaded a possible fix, I had to remove the window ReloadRoot() on scene closed and also fixed to not store references to hierarchy objects in the asset. Not sure why it was failing though, I understand it is a multiscene related issue and something probably related with having references to things being destroyed or something. I found similar issues in the unity forum.

Hope to have it fixed now. I will also disable recording hierarchy objects while application is playing to avoid other issues.
